### PR TITLE
update tiflash documents about cartesian join support

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
update tiflash documents about cartesian join

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6429
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/use-tiflash.md [hunk 1]
```diff
@@ -226,10 +226,12 @@ TiFlash supports the push-down of the following operators:
 * TopN: Performs the TopN calculation.
 * Limit: Performs the limit calculation.
 * Project: Performs the projection calculation.
-* HashJoin: Performs the join calculation based on the [Hash Join](/explain-joins.md#hash-join) algorithm, but with the following conditions:
+* HashJoin (having the join condition with the equivalent value): Performs the join calculation based on the [Hash Join](/explain-joins.md#hash-join) algorithm, but with the following conditions:
     * The operator can be pushed down only in the [MPP mode](#use-the-mpp-mode).
-    * The operator must have the join condition with the equivalent value.
     * The push-down of `Full Outer Join` is not supported.
+* Hashjoin (Cartesian Join, which does not have the join condition with the equivalent value): Performs the Cartesian Join algorithm, but with the following conditions:
```
Comment (Joyinqin) [inline tiflash/use-tiflash.md:232]:
```suggestion
* HashJoin (Cartesian Join, which does not have the join condition with the equivalent value): Performs the Cartesian Join algorithm, but with the following conditions:
```

**Review by Joyinqin — APPROVED** [review]
rest lgtm

File: tiflash/use-tiflash.md [hunk 2]
```diff
@@ -226,10 +226,12 @@ TiFlash supports the push-down of the following operators:
 * TopN: Performs the TopN calculation.
 * Limit: Performs the limit calculation.
 * Project: Performs the projection calculation.
-* HashJoin: Performs the join calculation based on the [Hash Join](/explain-joins.md#hash-join) algorithm, but with the following conditions:
+* HashJoin (having the join condition with the equivalent value): Performs the join calculation based on the [Hash Join](/explain-joins.md#hash-join) algorithm, but with the following conditions:
```
Comment (windtalker) [inline tiflash/use-tiflash.md:229]: HashJoin(Equi Join)

File: tiflash/use-tiflash.md [hunk 3]
```diff
@@ -226,10 +226,12 @@ TiFlash supports the push-down of the following operators:
...
     * The operator can be pushed down only in the [MPP mode](#use-the-mpp-mode).
-    * The operator must have the join condition with the equivalent value.
     * The push-down of `Full Outer Join` is not supported.
+* HashJoin (Cartesian Join, which does not have the join condition with the equivalent value): Performs the Cartesian Join algorithm, but with the following conditions:
```
Comment (windtalker) [inline tiflash/use-tiflash.md:232]: HashJoin(Non-Equi Join)

File: tiflash/use-tiflash.md (hunk 2 above)
Comment (TomShawn) [reply tiflash/use-tiflash.md:229]:
```suggestion
* HashJoin (Equi Join): Performs the join calculation based on the [Hash Join](/explain-joins.md#hash-join) algorithm, but with the following conditions:
```

File: tiflash/use-tiflash.md (hunk 3 above)
Comment (TomShawn) [reply tiflash/use-tiflash.md:232]:
```suggestion
* HashJoin (Non-Equi Join): Performs the Cartesian Join algorithm, but with the following conditions:
```

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**